Assignments/Sorting/ShellSort.playground/Pages/Shell Sort.xcplaygroundpage/Contents.swift

```diff
@@ -51,14 +58,29 @@ class KnuthSequence: SequenceType {
     
     func generate() -> AnyGenerator<Int> {
         
+        var index = 1
+        
+        while (3 ** index - 1) / 2 < self.n {
+            index += 1
+        }
+        
+        index -= 1
+        
+        var number = (3 ** index - 1) / 2
+        
         return anyGenerator {
-            nil
+            print("index: \(index)")
+            print("number: \(number)")
+            number = (3 ** index - 1) / 2
+            index -= 1
+            return number > 1 ? number : nil
```

Comment:
You never generate the last gap value, which must be equal to `1`, because the last step of shell sort is an insertion sort. Try to change from 
```
return number > 1 ? number : nil
```

to 

```
return number >= 1 ? number : nil
```
or 
```
return number > 0 ? number : nil
```